File: packages/firebase_app_check/firebase_app_check/android/src/main/java/io/flutter/plugins/firebase/appcheck/TokenChannelStreamHandler.java
/*
 * Copyright 2022, the Chromium project authors.  Please see the AUTHORS file
 * for details. All rights reserved. Use of this source code is governed by a
 * BSD-style license that can be found in the LICENSE file.
 */

package io.flutter.plugins.firebase.appcheck;

import com.google.firebase.appcheck.FirebaseAppCheck;
import io.flutter.plugin.common.EventChannel;
import java.util.HashMap;
import java.util.Map;

public class TokenChannelStreamHandler implements EventChannel.StreamHandler {

  private final FirebaseAppCheck firebaseAppCheck;
  private FirebaseAppCheck.AppCheckListener listener;

  public TokenChannelStreamHandler(FirebaseAppCheck firebaseAppCheck) {
    this.firebaseAppCheck = firebaseAppCheck;
  }

  @Override
  public void onListen(Object arguments, EventChannel.EventSink events) {

    listener =
        result -> {
          Map<String, Object> event = new HashMap<>();
          event.put("token", result.getToken());
          events.success(event);
        };

    firebaseAppCheck.addAppCheckListener(listener);
  }

  @Override
  public void onCancel(Object arguments) {
    if (listener != null) {
      firebaseAppCheck.removeAppCheckListener(listener);
      listener = null;
    }
  }
}

File: packages/firebase_app_check/firebase_app_check/android/src/main/kotlin/io/flutter/plugins/firebase/appcheck/TokenChannelStreamHandler.kt
// Copyright 2025 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
package io.flutter.plugins.firebase.appcheck

import com.google.firebase.appcheck.FirebaseAppCheck
import io.flutter.plugin.common.EventChannel

class TokenChannelStreamHandler(
  private val firebaseAppCheck: FirebaseAppCheck
) : EventChannel.StreamHandler {

  private var listener: FirebaseAppCheck.AppCheckListener? = null

  override fun onListen(arguments: Any?, events: EventChannel.EventSink) {
    listener = FirebaseAppCheck.AppCheckListener { result ->
      val event = HashMap<String, Any?>()
      event["token"] = result.token
      events.success(event)
    }
    firebaseAppCheck.addAppCheckListener(listener!!)
  }

  override fun onCancel(arguments: Any?) {
    listener?.let {
      firebaseAppCheck.removeAppCheckListener(it)
      listener = null
    }
  }
}
